### ISSUE TITLE
Use SVG icon

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/date_ical#18]
It would be good to update the icon usage from the dated raster images to SVG.

**Comment by laryn** [comment]
In addition to moving to SVG, for accessibility reasons let's make it 44x44px so the target is big enough:

https://www.w3.org/WAI/WCAG21/Understanding/target-size.html

**Comment by laryn** [comment]
@aitala or @gto1 Are you interested in testing this PR?

**Comment by gto1** [comment]
I *think* what should happen here is that the old orange png should now display as the black svg icon? I probably need to tweak something but I'm still seeing the old ical icon. I set up one of the Backdrop sites (success), git cloned this module repo to another location locally, switched to the fix branch, manually swapped that into my local ddev site in modules/contrib/date_ical, then created a test content type with the Date field, added a test node, created a test view (as Page and Feed) to see it, cleared cache, etc. all of which is doing as expected here (screenshot) except for the old png ical icon is still there. 

<img width="1059" height="878" alt="Image" src="https://github.com/user-attachments/assets/ef77f5fe-0a86-42c1-8184-a6bf58201a80" />

**Comment by gto1** [comment]
I prob didn't set my test content up right, the ical icon in my test looks to be coming from Viewsicalexport module: 
https://success.ddev.site/modules/contrib/viewsicalexport/images/ical-feed-icon.png

**Comment by aitala** [comment]
> [@aitala](https://github.com/aitala) or [@gto1](https://github.com/gto1) Are you interested in testing this PR?

Is there a way to download a patch for this...?

Thanks,
Eric

**Comment by aitala** [comment]
OK, I got it on a dev site... 

Is the module not loading the CSS file?

<img width="929" height="962" alt="Image" src="https://github.com/user-attachments/assets/cd27a13e-7e95-4986-b6c8-da5311516e3e" />

**Comment by aitala** [comment]
Yes, it needs 

`stylesheets[all][] = css/date_ical.css
`

**Comment by laryn** [comment]
@aitala I'm using the new-ish technique of attaching the CSS in the hook_theme definition so that it's only loaded when the theme definition is used rather than always being shoved in -- can you try clearing caches on the dev site to see if that triggers the theme definition rebuild and solves it?

**Comment by laryn** [comment]
@gto1 I filed an issue upstream in the `viewsicalexport` module as well in case the maintainer wants to use a similar approach there.

**Comment by aitala** [comment]
I pulled the MR down, removed the files from the dev site, and put in the new MR...

Still big SVG

https://dev.urfm.psu.edu/events-and-deadlines

Even after cache clears.

**Comment by indigoxela** [comment]
Something you may not be aware of is the fact that both modules, date_ical and viewsicalexport use a _theme_ function for displaying the icon.

So any theme can override it already with MYTHEME_date_ical_icon() or MYTHEME_viewsicalexport_icon(). This is something to keep in mind, when changing this module's theme function and especially when attaching new CSS to it. Make sure, it's backwards compatible and doesn't break in existing sites. 😉

**Comment by laryn** [comment]
The issue with the CSS not appearing for @aitala may just be an outdated core on his dev machine. 

Fair point about the theme function, but both of these fixes (SVG and A11Y) are things I would like to make default rather than assuming everyone will override the theme if they want the improvements. Perhaps it warrants a major point upgrade with this change, so the upgrade process has a little more friction for existing users to take note of the changes, but the updated version is provided by default for new users.

**Comment by indigoxela** [comment]
@laryn I didn't mean to stop or discourage here in any way. My comment only means "caution, test before merge / release, that this doesn't have weird side effects". 😉

**Comment by aitala** [comment]
As far as I can tell this has no negative effects - and I did get https://dev.urfm.psu.edu/events-and-deadlines working

Thanks, Eric
